packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -356,11 +356,10 @@ class _TextLayout {
     // Luckily they have the same bidi embedding level as the paragraph as per
     // https://unicode.org/reports/tr9/#L1, so we can anchor the caret to the
     // last logical trailing space.
-    final bool hasTrailingSpaces = switch (rawString.codeUnitAt(rawString.length - 1)) {
-      0x9 ||        // horizontal tab
-      0x3000 ||     // ideographic space
-      0x20 => true, // space
-      _ => false,
+    final String lastCharactor = rawString[rawString.length - 1];
```

Comment:
This is still checking the last code unit. What's the difference between this and the expression before the change?

Reply:
Also I'm curious how the regexp fares with surrogate pairs, could you also test a surrogate pair ("\uD801\uDC37" for example)?

Reply:
Ah I tested locally and it didn't throw. So up to you if you want to skip adding that test.